huaweicloud/services/dcs/resource_huaweicloud_dcs_custom_template.go

```diff
@@ -0,0 +1,432 @@
+		return nil
+	}
+	paramsMap := buildParamsMap(d)
+	curJson := utils.PathSearch("params", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		paramName := utils.PathSearch("param_name", v, "").(string)
+		if !paramsMap[paramName] {
+			continue
+		}
+		rst = append(rst, map[string]interface{}{
+			"param_id":      utils.PathSearch("param_id", v, nil),
+			"param_name":    paramName,
+			"param_value":   utils.PathSearch("param_value", v, nil),
+			"default_value": utils.PathSearch("default_value", v, nil),
+			"value_range":   utils.PathSearch("value_range", v, nil),
+			"value_type":    utils.PathSearch("value_type", v, nil),
+			"description":   utils.PathSearch("description", v, nil),
+			"need_restart":  utils.PathSearch("need_restart", v, nil),
+		})
+	}
+	return rst
+}
+
+func buildParamsMap(d *schema.ResourceData) map[string]bool {
+	params := d.Get("params").(*schema.Set).List()
+	paramsMap := make(map[string]bool)
+	for _, param := range params {
+		if v, ok := param.(map[string]interface{}); ok {
+			paramsMap[v["param_name"].(string)] = true
+		}
+	}
+	return paramsMap
+}
+
+func resourceCustomTemplateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateCustomTemplateChanges := []string{
+		"name",
+		"description",
+		"params",
+	}
+
+	if d.HasChanges(updateCustomTemplateChanges...) {
+		// updateCustomTemplate: update DCS custom template
+		var (
+			updateCustomTemplateHttpUrl = "v2/{project_id}/config-templates/{template_id}"
+			updateCustomTemplateProduct = "dcs"
+		)
+		updateCustomTemplateClient, err := cfg.NewServiceClient(updateCustomTemplateProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating DCS client: %s", err)
+		}
+
+		updateCustomTemplatePath := updateCustomTemplateClient.Endpoint + updateCustomTemplateHttpUrl
+		updateCustomTemplatePath = strings.ReplaceAll(updateCustomTemplatePath, "{project_id}",
+			updateCustomTemplateClient.ProjectID)
+		updateCustomTemplatePath = strings.ReplaceAll(updateCustomTemplatePath, "{template_id}", d.Id())
+
+		updateCustomTemplateOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+		}
+
+		updateCustomTemplateOpt.JSONBody = utils.RemoveNil(buildUpdateCustomTemplateBodyParams(d))
+		_, err = updateCustomTemplateClient.Request("PUT", updateCustomTemplatePath, &updateCustomTemplateOpt)
+		if err != nil {
+			return diag.Errorf("error updating DCS custom template: %s", err)
+		}
+	}
+	return resourceCustomTemplateRead(ctx, d, meta)
+}
+
+func buildUpdateCustomTemplateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
```

Comment:
support clear description

```suggestion
		"description": d.Get("description"),
```